pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -304,7 +304,13 @@ func (proxy *DataSourceProxy) validateRequest() error {
 			continue
 		}
 
-		if route.ReqRole.IsValid() {
+		routeEval := route.Evaluator()
```

Comment:
Would it be possible to reuse `hasAccessToRoute` here?

If not, I think RBAC access evaluator should be behind a feature toggle, the same way it is for `hasAccessToRoute`.

Reply:
I didn't see a nice way to share the logic between the two, but I ported the same logic over: b10cbb7b30fba2476fb72c18f5d9973ee1d17778